libkbfs/journal_md_ops.go

```diff
@@ -403,6 +403,11 @@ func (j journalMDOps) GetRange(ctx context.Context, id tlf.ID, start,
 		j.jServer.deferLog.LazyTrace(ctx, "jMDOps: GetRange %s %d-%d done (err=%v)", id, start, stop, err)
 	}()
 
+	if lockBeforeGet != nil {
+		// We need to grab locks. So bypass journal and hit server directly.
+		return j.MDOps.GetRange(ctx, id, start, stop, lockBeforeGet)
```

Comment:
I don't think this is correct.  It will not return any MD entries that happen to be in the journal.  In your use case that's fine, because we only ever set the lock when fetching entries that come from the server, but in general it's dangerous if there are ever other types of callers.

I'd be more comfortable if we called the delegate MDOps directly from `folderBranchOps.getAndApplyMergedUpdates`, but that would require a bunch of annoying changes.  Maybe for now just return an if there's any revision in the journal within the requested range, when the lock is set?

Reply:
Oh I see; I didn't look closely enough and wrongly think this goes to server directly. Thanks for spotting that! 

>  Maybe for now just return an if there's any revision

Is it "just return an error if" what you meant?

Reply:
> I didn't look closely enough and wrongly think this goes to server directly

Your code here does go to the server directly.  But my point is that it's wrong to only go to the server if there's anything in the journal within the start/stop range.

And yeah, I meant "error".

Reply:
Oh I see you point now. Thanks for clarification! Error sounds good; will do!